### ISSUE TITLE
freetube 0.7.1 subscriptions loading erratically [bug]

### ISSUE THREAD

**Issue by mcmurphy8097234789** [FreeTubeApp/FreeTube#430]
**Expected Behaviour:**
Freetube appimage boots, loads subscriptions, and refreshing of subscriptions is allowed to occur 

**Actual behaviour:**
When Freetube Appimage is loaded initially, or by clicking 'refresh,' channels load erratically. Some will load, others will not. Which ones load seems to be somewhat random. Never are all of them loaded.

**Parameters:**

This is the same regardless of whether proxying through invidio.us is used or not or whether videos are grabbed locally. A Tor/proxy is NOT used for API calls. Redownloading appimage does not change outcome.

Bug is reproducible on different computers on different internet networks with different IP addresses. This leads me to believe this is a problem with a function of the program, and not the current hardware or network being used. 

If requested, will try to provide subscription text and debug log

Edit with OS information:
OS: Solus 4.0 fortitude
Kernel: x86_64 Linux 5.3.10-134.current

**Comment by Omelettechu** [comment]
This issue has also been affecting me now for a few days, have not tried on other systems, but I have a couple of other machines I can potentially try this on during the weekend.

I have been on 0.7.1 since release and this has only started happening this week.
I haven't looked at the codebase, but could it possibly be related to a youtube API change of some sort?

Here is my OS info just in case:
Windows 10 Version 1809 Build 17763.379

**Comment by LWFlouisa** [comment]
Same thing seems to be happening on this end.

OS: Linux Mint 19.1 I'm unsure of what specific build. It's an Intel Core 3.0 I think.

**Comment by linuxgirl22** [comment]
Same here (arch zen installer/mate), and just started over the last week ... 3rd time needing to change invidiou.us instance, but keeps breaking. Worrying that no longer any dev responses in Riot general chat.

**Comment by LWFlouisa** [comment]
My issue seems to be caused by Invidio.us instance being blocked, going to test out another instance of invidio and see if that resolves the issue.

**Comment by Omelettechu** [comment]
This seems to be caused by blocked invidious instances (see [this reddit thread](https://www.reddit.com/r/privacy/comments/e22bv6/seems_like_all_official_invidious_instances_are/)). It looks like freetube proxies everything through invidous even if you haven't enabled the option. I was unaware of this and thought it was accessing youtube directly (a la newPipe). Changing to a not-blocked instance should solve this (https://invidiou.sh was recommended in the thread and works for me).

**Comment by linuxgirl22** [comment]
No problem @Omelettechu and, yes, I just discovered the settings make no difference, after discovering I didn't even have the invidio.us option turned on. Have tried various instances and various settings, but Freetube seems too broken at this point.

**Comment by linuxgirl22** [comment]
Today, invidio.us itself is finding and playing videos great, but, whether with 'proxy through invidio.us' turned on or off, Freetube loads nothing. I'd thought invidio.us itself wouldn't load, if instances were breaking, but it's fine, so maybe it's something more Freetube-related?

**Comment by Omelettechu** [comment]
Nice find, it's probably safe to say in this case that the issue is being cause by the fact that freetube does everything through an invidous instance, so the problem actually lies within the fact that invidous instances are being struck down and/or limited access by google. AFAIK this seems to happen from time to time with invidous instances and they usually end up recovering from it, but I think a better, more long term solution, would be to have the option to access youtube APIs directly (like other youtube frontends eg: newPipe) instead of proxying through invidous for when things like this happen.

Of course the permanent solution would be getting everyone on to a decentralized video platform, such as PeerTube, but I don't see that happening for a few years (would be nice though hahah).

**Comment by linuxgirl22** [comment]
Yes, good to find that instance (Trisquel forum, I think) ... at the moment, lol, working well in the browser, but not Freetube (loads subs, but error when click through to a video), but, as you say, it can be very up and down, or some channels work on one instance, some on another. 

Sometimes needing to fall back on GTK youtube-viewer at the moment (YT api, eek), if neither browser instance works, but it's good to hear that things end up fixed okay. 

A button to switch to YT api could be cool, and decentralised would be awesome!

**Comment by LWFlouisa** [comment]
I wonder if the router has somewhat to do with it. I ask as moving to a different network I didn't have the same problems.

But I can find my profile again, and listen to music playlists.

**Comment by Omelettechu** [comment]
@LWFlouisa It's probably not the router as everything (minus loading the actual video) is through the invidious API as stated at freetubeapp.io:

>  FreeTube is a YouTube client for Windows, Mac, and Linux built around using YouTube more privately. You can enjoy your favorite content and creators without your habits being tracked. All of your user data is stored locally and never sent or published to the internet. **Being powered by the Invidious API**, FreeTube has become one of the best methods to watch YouTube privately on desktop.

If the issue has gone away then that most likely means whichever invidious instance you were using (invidio.us by default) has solved its' youtube connectivity issues.

Also FYI for anyone following me trying to make sense of how connections to youtube are made, I actually read things this time, the proxy videos toggle actually does serve a purpose and is part of the invidious API.

Basically with the invidious API everything _except_ watching a video is proxied through an invidious instance by default. So this means once you start loading a video, your accessing youtube directly via googlevideo.com. When you enable video proxying through invidious, instead of accessing googlevideo.com to fetch the video, you make to the request to the invidious instance which then makes the request to googlevideo.com.

Also I spelled invidious wrong in my previous posts, too lazy to go edit all that hahah.

**Comment by linuxgirl22** [comment]
Thanks for explaining exactly how Freetube works ... everything but vids was playing, and now, via switching on 'proxy through invidious', they're working.

**Comment by LWFlouisa** [comment]
Still happening. Is there not a way to regulate this instance blocking thing? This out of control instance blocking is one the reasons I no longer take Mastodon seriously as a fediverse platform.

Or at least, some way to quickly work around it.

This kind of thing never happens on Diaspora. Not sure why Invidio is any different.

It shouldn't matter if an instance is in Denmark or Japan, it should operate on the same general guidelines if it's going to enforce them, otherwise you have people blocked for largely arbitrary reasons: in practice extremely authoritarian. And not consistently so.

**Comment by Omelettechu** [comment]
Just to be clear, any invidious instance does not have control on whether it gets blocked or not. Invidious scrapes the youtube API and is an unofficial front end for youtube. 

Google does not endorse and is probably very against these unofficial apps scraping their web APIs. In the case of newPipe on android, all requests to youtube servers are made via your device, thus exposing your IP address to Google. Since Google is still able to mine some sort of information from you (your IP address), they would disavow such an app, but not go out of their way to shut it down.

In the case of Invidious instances, the only requests that _could_ be linked back to your IP address via the google video server URL to fetch the video data. Now it could also be the case that users on an instance enable the "Proxy Videos" option, which would then eliminate any tracking of your IP address by Google (in terms of youtube usage).

Knowing how Invidious instances actually provide content from youtube to the end user is important to understand why the instances get blocked. 

Since Google has started limiting requests from any single IP address to their youtube API, this means that Invidious instances, which would have all requests except for possibly the actual fetching of video data, are potentially sending out enough requests to be seen as a bad client by Google and subsequently blocked for spamming their API. Couple that with fact that many users on invidious most likely proxy the videos through the instance too, and it becomes more certain that an instance will be blocked.

A normal individual user would probably get nowhere near the amount of requests required to be put on the blocklist, so since apps like newPipe make requests via your own IP address for everything, such apps would not be affected by this.

So the invidious instances are not at fault here, it's all Google trying to have a monopoly on everything due to their insane fetish of data mining everyone.

TL;DR: Invidious instances aren't at fault, it's scumbag Google doing the blocking.
